pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StEqualsFunction.java

```diff
@@ -53,14 +54,16 @@ public void init(List<TransformFunction> arguments, Map<String, DataSource> data
     TransformFunction transformFunction = arguments.get(0);
     Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
         "First argument must be single-valued for transform function: %s", getName());
-    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES,
-        "The first argument must be of bytes type");
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        notBytesErrorMessage(transformFunction, "first"));
     _firstArgument = transformFunction;
     transformFunction = arguments.get(1);
     Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
         "Second argument must be single-valued for transform function: %s", getName());
-    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES,
-        "The second argument must be of bytes type");
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        notBytesErrorMessage(transformFunction, "second"));
```

Comment:
Let's not eagerly create strings in checks like this - instead use the template overload and provide parameters, or don't use `Preconditions` and move the construction of the error message within the conditions so that it is only constructed in exceptional cases.